### PR TITLE
Recovery password

### PR DESCRIPTION
**Pull request by panel14**

### CODE REVIEW

**Comment by IP-13** [comment]
|Overall Project|65.08% **`-2.8%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|3.08%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/f4c374ccf68a18e58af896e0451600a2a82c33c6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|77.27% **`-13.64%`**|:x:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/f4c374ccf68a18e58af896e0451600a2a82c33c6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|66.96% **`-14.35%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/f4c374ccf68a18e58af896e0451600a2a82c33c6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|18.99% **`-21.2%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/f4c374ccf68a18e58af896e0451600a2a82c33c6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|18.92% **`-36.49%`**|:x:|
|[UserPasswordRecoveryInfoService.java](https://github.com/itmo-events-app/event-app/blob/f4c374ccf68a18e58af896e0451600a2a82c33c6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserPasswordRecoveryInfoService.java)|14.63% **`-85.37%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|64.95% **`-2.8%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|3.08%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/da4818588a62b710b93f4a316b3282940e989d02/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|77.27% **`-13.64%`**|:x:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/da4818588a62b710b93f4a316b3282940e989d02/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|66.96% **`-14.35%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/da4818588a62b710b93f4a316b3282940e989d02/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|18.99% **`-21.2%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/da4818588a62b710b93f4a316b3282940e989d02/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|18.92% **`-36.49%`**|:x:|
|[UserPasswordRecoveryInfoService.java](https://github.com/itmo-events-app/event-app/blob/da4818588a62b710b93f4a316b3282940e989d02/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserPasswordRecoveryInfoService.java)|14.63% **`-85.37%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|64.95% **`-2.8%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|3.08%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/bd67e2c7d57a8ed0c313de1768254bea7aee57bf/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|77.27% **`-13.64%`**|:x:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/bd67e2c7d57a8ed0c313de1768254bea7aee57bf/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|66.96% **`-14.35%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/bd67e2c7d57a8ed0c313de1768254bea7aee57bf/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|18.99% **`-21.2%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/bd67e2c7d57a8ed0c313de1768254bea7aee57bf/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|18.92% **`-36.49%`**|:x:|
|[UserPasswordRecoveryInfoService.java](https://github.com/itmo-events-app/event-app/blob/bd67e2c7d57a8ed0c313de1768254bea7aee57bf/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserPasswordRecoveryInfoService.java)|14.63% **`-85.37%`**|:x:|

File: src/main/java/org/itmo/eventapp/main/model/dto/request/NewPasswordRequest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.itmo.eventapp.main.model.dto.request;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotBlank;
+import org.itmo.eventapp.main.model.validation.annotation.PasswordMatching;
+import org.itmo.eventapp.main.model.validation.annotation.StrongPassword;
+
+@SuppressWarnings("java:S2068")
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/model/dto/request/NewPasswordRequest.java:8]: а что это за предупреждение?

Comment (panel14) [reply src/main/java/org/itmo/eventapp/main/model/dto/request/NewPasswordRequest.java:8]: Про то что пароль какой-то явно в коде используется
А это просто название поля

File: src/main/java/org/itmo/eventapp/main/controller/AuthController.java [hunk 2]
```diff
@@ -61,4 +63,28 @@ ResponseEntity<List<RegistrationRequestForAdmin>> listRegisterRequests() {
                 .status(HttpStatus.OK)
                 .body(authenticationService.listRegisterRequestsCallback());
     }
+
+    @Operation(summary = "Отправка запроса на восстановление пароля")
+    @PostMapping("/recoveryPassword")
+    ResponseEntity<Void> recoveryPassword(@Valid @RequestBody RecoveryPasswordRequest request) {
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/controller/AuthController.java:69]: Если ничего не возвращается - 204 статус, NoContent

File: src/main/java/org/itmo/eventapp/main/model/dto/request/NewPasswordRequest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        @Schema(example = "c5b7bcc0-cffe-4f57-853c-7fa18e56b36d", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
+        String token,
+
+        @NotBlank(message = "Поле пароля не должно быть пустым")
+        @StrongPassword
+        @Schema(example = "PaSsWoRd1", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
+        String newPassword,
+
+        @NotBlank(message = "Поле подтверждения пароля не должно быть пустым")
+        @Schema(example = "PaSsWoRd1", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/model/dto/request/NewPasswordRequest.java:24]: NotBlank и NOT_REQUIRED противоречат друг другу

File: src/main/java/org/itmo/eventapp/main/service/AuthenticationService.java [hunk 4]
```diff
@@ -154,4 +160,37 @@ public List<RegistrationRequestForAdmin> listRegisterRequestsCallback() {
...
+
+    public void recoverPassword(String email, String returnUrl) {
+
+        UserLoginInfo info = userLoginInfoService.findByLogin(email);
+
+        if (info.getLoginStatus() != LoginStatus.APPROVED) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT, ExceptionConst.EMAIL_NOT_APPROVED);
+        }
+
+        User user = userService.findByLogin(email);
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/service/AuthenticationService.java:172]: Можно взять из info, чтобы второй раз в БД не ходить, он EAGER подгружается

File: src/main/resources/application-local.yml [hunk 5]
```diff
@@ -17,10 +17,10 @@ spring:
   mail:
     host: smtp.mail.ru
     port: 465
-    username: username # нерабочая конфигурация, нужны реальные имя и пароль в env
-    password: password # обращаться за данными к @alex_darklol
-#    username: ${EMAIL_USERNAME}
-#    password: ${EMAIL_PASSWORD}
+    #username: username # нерабочая конфигурация, нужны реальные имя и пароль в env
+    #password: password # обращаться за данными к @alex_darklol
+    username: ${EMAIL_USERNAME}
```
Comment (IP-13) [inline src/main/resources/application-local.yml:22]: а зачем тут ща менять, надо в конфигурацию прописывать переменные, верни пока это

File: src/main/resources/db/migration/V0_2__create_tables.sql [hunk 6]
```diff
@@ -123,6 +123,14 @@ create table if not exists task
     reminder timestamp not null
 );
 
+create table if not exists user_password_recovery_info
```
Comment (IP-13) [inline src/main/resources/db/migration/V0_2__create_tables.sql:126]: В миграции с тестами тоже надо добавить это

**Comment by IP-13** [comment]
|Overall Project|63.61% **`-2.67%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|3.11%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/0343d289d77aa65116cda27aa3986a3917ac25eb/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|69.8% **`-11.02%`**|:x:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/0343d289d77aa65116cda27aa3986a3917ac25eb/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|66.96% **`-14.35%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/0343d289d77aa65116cda27aa3986a3917ac25eb/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|19.05% **`-20.63%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/0343d289d77aa65116cda27aa3986a3917ac25eb/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|18.92% **`-36.49%`**|:x:|
|[UserPasswordRecoveryInfoService.java](https://github.com/itmo-events-app/event-app/blob/0343d289d77aa65116cda27aa3986a3917ac25eb/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserPasswordRecoveryInfoService.java)|14.63% **`-85.37%`**|:x:|

File: src/main/java/org/itmo/eventapp/main/controller/AuthController.java (hunk 2 above)
Comment (IP-13) [reply src/main/java/org/itmo/eventapp/main/controller/AuthController.java:69]: В других методах тоже 204 статус

**Comment by IP-13** [comment]
|Overall Project|63.61% **`-2.67%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|3.11%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/dc3dd1c5fae35de36219e2781a135612f91a0804/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|69.8% **`-11.02%`**|:x:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/dc3dd1c5fae35de36219e2781a135612f91a0804/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|66.96% **`-14.35%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/dc3dd1c5fae35de36219e2781a135612f91a0804/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|19.05% **`-20.63%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/dc3dd1c5fae35de36219e2781a135612f91a0804/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|18.92% **`-36.49%`**|:x:|
|[UserPasswordRecoveryInfoService.java](https://github.com/itmo-events-app/event-app/blob/dc3dd1c5fae35de36219e2781a135612f91a0804/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserPasswordRecoveryInfoService.java)|14.63% **`-85.37%`**|:x:|

**Review by omgronny — APPROVED** [review]

**Comment by IP-13** [comment]
|Overall Project|61.16% **`-2.73%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|2.9%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/42c5569c0c4e46e6d453f8171d87d4df6c2beffa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|67.35% **`-11.02%`**|:x:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/42c5569c0c4e46e6d453f8171d87d4df6c2beffa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|62.35% **`-19.03%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/42c5569c0c4e46e6d453f8171d87d4df6c2beffa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|15.79% **`-16.29%`**|:x:|
|[UserPasswordRecoveryInfoService.java](https://github.com/itmo-events-app/event-app/blob/42c5569c0c4e46e6d453f8171d87d4df6c2beffa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserPasswordRecoveryInfoService.java)|14.63% **`-85.37%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/42c5569c0c4e46e6d453f8171d87d4df6c2beffa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|14.43% **`-27.84%`**|:x:|

**Review by IP-13 — APPROVED** [review]

**Pull request merged**